### PR TITLE
Feature/filter

### PR DESCRIPTION
**Pull request by reald002**

### CODE REVIEW

**Comment by IslamRustamov1** [comment]
resolve conflicts please

File: src/components/Filters/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    const handleClearBtnClick = () => {
+        props.onClearBtnClick();
+    }
+
+    return (
+        <div className="filters">
+            <div className='items-left filters__item'>
+                {props.data.filter(e => !e.checked).length > 0 ? `${props.data?.filter(e => !e.checked).length} item${props.data?.filter(e => !e.checked).length > 1 ? 's' : ''} left` : ''}
```
Comment (IslamRustamov1) [inline src/components/Filters/index.js]: 3 repetitive filters in one line

File: src/components/Filters/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+function Filters(props) {
+    const filters = [
+        {text: 'All'},
+        {text: 'Active'},
+        {text: 'Completed'}
+    ];
+
+    const handleRadioChange = (e) => {
+        const value = e.target.value === 'Active' ? false : e.target.value === 'Completed' ? true : e.target.value;
```
Comment (IslamRustamov1) [inline src/components/Filters/index.js]: Try to avoid this kind of multiple ternary operators. Better to rewrite it with switch.

**Review by IslamRustamov1 — APPROVED** [review]

File: src/App.js [hunk 3]
```diff
@@ -24,13 +27,58 @@ class App extends React.Component {
         }));
     }
 
+    handleCheckboxClick = (value, id) => {
+        const newObj = Object.assign(this.state.data.find(e => e.id === id));
```
Comment (AnnaKulkova) [inline src/App.js]: You'll get an error if **find** returns **undefind**.

And don't use assign to copy object, use spread operator instead.

File: src/App.js [hunk 4]
```diff
@@ -24,13 +27,58 @@ class App extends React.Component {
...
+    }
+
+    handleClearBtnClick = () => {
+        this.setState(state => ({
+            data: state.data.filter(e => !e.checked)
+        }));
+    }
+
+    handleToggleAllClick = () => {
+        const todos = this.checkbox.current.children;
```
Comment (AnnaKulkova) [inline src/App.js:52]: You have to use state, not refs. You avoid react philosophy

File: src/components/Filters/filter-btn.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+function FilterBtn({id, text}) {
+    return (
+        <span>
+            <input defaultChecked={id === 0 ? 'checked': ''} id={`filter-${id}`} value={text} className="filter-radio" name="filter-radio" type="radio"/>
```
Comment (AnnaKulkova) [inline src/components/Filters/filter-btn.js]: What do you mean by `id === 0`?

File: src/components/Filters/index.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    const getUncheckedTodosLength = () => {
+        return props.data.filter(e => !e.checked).length;
+    }
+
+    return (
+        <div className="filters">
+            <div className='items-left filters__item'>
+                { getUncheckedTodosLength() > 0 ? `${getUncheckedTodosLength()} item${getUncheckedTodosLength() > 1 ? 's' : ''} left` : ''}
+            </div>
+            <div onChange={handleRadioChange} className="filter-buttons filters__item">
```
Comment (AnnaKulkova) [inline src/components/Filters/index.js:28]: Are you sure that you want to pass it into div? Not in Button?

File: src/components/Filters/index.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+function Filters(props) {
+    const filters = [
+        {text: 'All'},
+        {text: 'Active'},
+        {text: 'Completed'}
+    ];
+
+    const handleRadioChange = (e) => {
+        let value;
+        switch(e.target.value) {
```
Comment (AnnaKulkova) [inline src/components/Filters/index.js]: Again, use state, not target

File: src/components/TodoItem/index.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        this.props.onRemoveTodo(this.props.id);
+    };
+
+    handleCheckboxClick = (e) => {
+        this.props.onCheckboxClick(e.target.checked, this.props.id);
+    };
+
+    render() {
+        return (
+            <li style={{ display: (this.props.filter === this.props.checked || this.props.filter === 'All') ? 'flex' : 'none' }}  className="todo-item">
```
Comment (AnnaKulkova) [inline src/components/TodoItem/index.js]: Use Conditional Rendering for it

**Review by AnnaKulkova — CHANGES_REQUESTED** [review]

**Review by AnnaKulkova — APPROVED** [review]
Not ideal but much better now. Merge when ready

**Pull request merged**